cs/cs/Serializer/VowpalWabbitJsonSerializer.cs

```diff
@@ -0,0 +1,298 @@
+                                if (propertyName.StartsWith("_"))
+                                {
+                                    // special fields
+                                    switch (propertyName)
+                                    {
+                                        case "_label":
+                                            this.ParseLabel(context);
+                                            break;
+                                        //case "_shared":
+                                        //    break;
+                                        //case "_actions":
+                                        //    break;
+                                        //case "action":
+                                        //    break;
+                                        default:
+                                            reader.Skip();
+                                            break;
+                                    }
+                                }
+                                else
+                                {
+                                    if (!reader.Read())
+                                        throw new ArgumentException("...");
+
+                                    if (reader.TokenType == JsonToken.StartObject)
+                                        this.ParseNamespaceAndFeatures(context, propertyName);
+                                    else
+                                        this.ParseFeature(defaultNamespaceContext, defaultNamespace, propertyName);
+
+                                }
+                                break;
+                        }
+                    }
+
+                    // append default namespaces features if we found some
+                    if (defaultNamespaceContext.StringExample != null && defaultNamespaceContext.StringExample.Length > 0)
+                    {
+                        context.StringExample.AppendFormat(CultureInfo.InvariantCulture,
+                            "| {0}", defaultNamespaceContext.StringExample);
+                    }
+                }
+
+                var vwExample = context.ExampleBuilder.CreateExample();
+
+                if (this.vw.Settings.EnableStringExampleGeneration)
+                    vwExample.VowpalWabbitString = context.StringExample.ToString();
+
+                return vwExample;
+            }
+        }
+
+        private void ParseLabel(VowpalWabbitMarshalContext context)
+        {
+            // peak the first property name
+            if (!reader.Read())
+                throw new ArgumentException("...");
+
+            if (reader.TokenType != JsonToken.StartObject)
+                throw new ArgumentException("...");
+
+            if (!reader.Read())
+                throw new ArgumentException("...");
+
+            if (reader.TokenType != JsonToken.PropertyName)
+                throw new ArgumentException("...");
+
```

Comment:
nit: you can potentially combine these 4 checks into a single statement or two separate ones i.e. if(!reader.Read() || tokenType != ...)